src/modules/users/users.repository.ts

```diff
@@ -0,0 +1,47 @@
+import { User, UserWithoutId } from './interfaces/user';
+import { UserSchema } from '../../schemas/users-schema';
+import { UserParameters } from './interfaces/parameters';
+
+class UserRepository {
+  async findByEmail(email: User['email']): Promise<User> {
+    const usersArr: Array<User> = await UserSchema.find({ email: email }).exec();
+
+    return usersArr[0] || null;
+  }
+
+  async findById(id: User['_id']): Promise<User | null> {
+    let user: User;
+    try {
+      user = await UserSchema.findById(id).exec();
+    } catch (error) {
+      user = null;
+    }
+    return user;
+  }
+
+  update(id, user: User): Promise<User> {
+    return UserSchema.findByIdAndUpdate(id, user, { new: true }).exec();
```

Comment:
what does the flag new do?

Reply:
It is for returning the document after update was applied.